### PR TITLE
vote create: support DRep and CC script hash

### LINKED ISSUE #661: Voting for script based CC members and script based DReps
# What
We need to expand `cardano-cli conway governance vote create` with the ability to produce votes for script-based governance actors "CC" and "DREPs"

`--cc-hot-script-hash`

and 

`--drep-script-hash `

So that we have: 

```
 cardano-cli conway governance vote create (--yes | --no | --abstain)
                                                   --governance-action-tx-id TXID
                                                   --governance-action-index WORD32
                                                   ( --drep-verification-key STRING
                                                   | --drep-verification-key-file FILE
                                                   | --drep-key-hash HASH
                                                   | --drep-script-hash 
                                                   | --stake-pool-verification-key STRING
                                                   | --cold-verification-key-file FILE
                                                   | --stake-pool-id STAKE_POOL_ID
                                                   | --cc-hot-verification-key STRING
                                                   | --cc-hot-verification-key-file FILE
                                                   | --cc-hot-key-hash STRING
                                                   | --cc-hot-script-hash HASH
                                                   )
                                                   [--anchor-url TEXT
                                                     --anchor-data-hash HASH]
                                                   --out-file FILE
```
# CDDL reference:

```
voting_procedures = { + voter => { + gov_action_id => voting_procedure } }

; Constitutional Committee Hot KeyHash: 0
; Constitutional Committee Hot ScriptHash: 1
; DRep KeyHash: 2
; DRep ScriptHash: 3
; StakingPool KeyHash: 4
voter =
  [ 0, addr_keyhash
  // 1, scripthash
  // 2, addr_keyhash
  // 3, scripthash
  // 4, addr_keyhash
  ]
```

### PR DESCRIPTION
**Pull request by smelc**
# Changelog

```yaml
- description: |
    vote create: support DRep and CC script hash
# uncomment types applicable to the change:
  type:
  - feature        # introduces a new feature
  # - breaking       # the API has changed in a breaking way
  - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
  # - improvement    # QoL changes e.g. refactoring
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

Fixes https://github.com/IntersectMBO/cardano-cli/issues/661

# Checklist

- [X] Commit sequence broadly makes sense and commits have useful messages
- [ ] Self-reviewed the diff

### CODE REVIEW

**Comment by CarlosLopezDeLara** [comment]
Perfect! 

Vote with Drep script hash 

`{[3, h'4A94737ECAA3238D7CAF94B87FC52D07FE978823D6D16C06F1163214']: {[h'66696EF1110A8F6574814DB66112CBEAFAA9C045143790278931D3AF6DBA3F41', 0]: [1, null]}}`

Vote with CC hot script hash

`{[1, h'4A94737ECAA3238D7CAF94B87FC52D07FE978823D6D16C06F1163214']: {[h'66696EF1110A8F6574814DB66112CBEAFAA9C045143790278931D3AF6DBA3F41', 0]: [1, null]}}`

**Comment by CarlosLopezDeLara** [comment]
Perfect! 

Vote with Drep script hash 

`{[3, h'4A94737ECAA3238D7CAF94B87FC52D07FE978823D6D16C06F1163214']: {[h'66696EF1110A8F6574814DB66112CBEAFAA9C045143790278931D3AF6DBA3F41', 0]: [1, null]}}`

Vote with CC hot script hash

`{[1, h'4A94737ECAA3238D7CAF94B87FC52D07FE978823D6D16C06F1163214']: {[h'66696EF1110A8F6574814DB66112CBEAFAA9C045143790278931D3AF6DBA3F41', 0]: [1, null]}}`

File: cardano-cli/src/Cardano/CLI/EraBased/Options/Common.hs [hunk 1]
```diff
@@ -3160,19 +3183,15 @@ pDRepHashSource =
...
 pConstitutionScriptHash :: Parser ScriptHash
 pConstitutionScriptHash =
-  Opt.option scriptHashReader $ mconcat
-    [ Opt.long "constitution-script-hash"
-    , Opt.metavar "HASH"
-    , Opt.help "Constitution script hash (hex-encoded). Obtain it with \"cardano-cli conway governance hash script ...\"."
-    ]
+  pScriptHash
+    "constitution-script-hash"
+    "Constitution script hash (hex-encoded). Obtain it with \"cardano-cli conway governance hash script ...\"."
```
Comment (CarlosLopezDeLara) [inline cardano-cli/src/Cardano/CLI/EraBased/Options/Common.hs]: conway governance hash anchor-data...

Comment (smelc) [reply cardano-cli/src/Cardano/CLI/EraBased/Options/Common.hs]: @CarlosLopezDeLara> I'm not sure what you mean. This help text seems correct (it is about the hash of a script).

**Review by carbolymer — APPROVED** [review]

**Comment by carbolymer** [comment]
Can you squash commits and improve their descriptions?

**Comment by smelc** [comment]
> Can you squash commits and improve their descriptions?

@carbolymer> done :+1:

**Review by Jimbo4350 — COMMENTED** [review]
:+1:

**Pull request merged**